### DISCUSSION TITLE
[iOS/Android] Add an option to decrease size of 2FA entries

### DISCUSSION THREAD

**Discussion by jimmyjones865** [ente-io/ente#1027] [Enhancements]
**Is your feature request related to a problem? Please describe.**
On my phone (iPhone SE 2020 / 4.7" screen), I can only see 5 entries at a time. Even if I only have ~ 20 entries, this leads to a lot of scrolling. In comparison, Mail app's inbox shows 12 emails per screen with sender and subject visible, and 9 emails if 1st line is previewed as well.

**Describe the solution you'd like**
Add an option to make entries smaller, allowing for more entries on screen. Trimming the padding on tiles should allow for 7-8 entries at least. 

**Describe alternatives you've considered**
Being able to pin frequently used entries to top as suggested in #134 would already help reduce the need for scrolling.
Smaller tiles could also be linked to the already existing option "Show large icons [off]" or "Hide codes [off]". Currently, neither of these options change the size of the overall tile.

Edit 17. Sept 24 to also include Android in title.

**Comment by lukasbestle** [comment]
I quite like the layout of [Raivo OTP](https://raivo-otp.com). It's very compact, yet clean.

**Comment by jimmyjones865** [comment]
I really appreciate the added 'pin' feature, thanks!
However, after the latest update the size of entries has increased even more. I can now only see 3 full and 1 half entry on my phone because the amount of white space has increased substantially. I did a quick and rough screenshot measurement that shows almost 60% of every entry's vertical size being white space, which is quite a lot. 
In my opinion, a reduction of white space and/or option to ~ half the vertical size of entries would be great.
This is a full, uncropped screenshot on my 4.7" iPhone 


![Bildschirmfoto 2024-05-21 um 09 59 35](https://github.com/ente-io/ente/assets/131913735/954de461-cacc-40f9-9a04-5f253cf1f3f5)

**Comment by sciamano89** [comment]
100% agree with this, I'd really like to have a "compact" view option. I'm on an iphone 13 pro and can only see 5 entries as well. No need to fix what's not broken, just an alternative display option that has smaller cells with reduced whitespace. Like @lukasbestle pointed above, it could have the name and details on the same line, and you could remove the "next" label. 

If you wanted to get fancy you could have the next totp animating from right to left, squeezing out the old code as it approaches the 30 seconds mark :) That could also remove the need for the colored bar, as visually the animation of the code (plus maybe coloring the code itself) would be sufficient.

**Comment by drcheap** [comment]
Came here to "feature request" this, but not just for iOS.  The desktop app (Windows in my case) is so large-and-in-charge that even a 2K monitor can only show 9 codes vertically.  That's 160 pixels per code whereas the actual text of a single code is about 18 pixels tall.  So much wasted space due to whitespace!

An option for compact mode, maybe similar to the design @lukasbestle referenced, would drastically improve the UX for those with a large quantity of codes.

P.S. All this giant font + whitespace is probably also contributing to the accessibility issues some people face (e.g. #2005 & #2590).

**Comment by mrx23dot** [comment]
Can we do this? Also for android.

We need more compact entries, 4/screen is insane, it makes it manually browsing a chore.

Have a look at Aegis, it has a more thought-through UI.

**Comment by drcheap** [comment]
The new Compact Mode in v4.0.0 is great!  Now shows about 8 codes with much less wasted space with the default window size.  Also, app window size (but unfortunately not position) is retained, so I can now get 12-14 codes visible right at startup w/o any further action. 👍

**Comment by mrx23dot** [comment]
6for me on android, I'm happy.

**Comment by sciamano89** [comment]
Looks great, exactly what I was looking for. Nice job!
